pkg/bootstrap/platform/gcp.go

```diff
@@ -144,21 +141,26 @@ func IsGCP() bool {
 // Metadata returned by the GCP Environment is taken from the GCE metadata
 // service.
 func NewGCP() Environment {
-	return &gcpEnv{}
+	return &gcpEnv{
+		fillMetadata: lazy.New(func() (bool, error) {
+			return shouldFillMetadata(), nil
```

Comment:
Should this be `e.shouldFIllMetadata`?

Or actually, isn't this a circular definition? It looks like gcpEnv.fillMetadata depends on shouldFillMetadata(), but that function depends on the existence of e.fillMetadata.

I may be missing something.

Reply:
`shouldFillMetadata()` can probably be inlined and then its less confusing

Reply:
Ah its because we mock it, forgot about that